src/components/Card/PokemonCard.tsx

```diff
@@ -1,43 +1,33 @@
-import { useNavigate } from "react-router-dom";
-import { PokiType } from "../../utils/Types";
-import { LazyLoadImage } from "react-lazy-load-image-component";
-// import "react-lazy-load-image-component/src/effects/blur.css";
 
+import { LazyLoadImage } from "react-lazy-load-image-component";
+import { getComplementaryColor } from "../../utils/typesColor";
 type cardProps = {
   id: number;
   name: string;
   imgURL: string;
   baseColor?: string;
   ability?: string;
-  type: PokiType;
+  type: PokiType[];
+  onClick: (id: number) => void;
 };
 
-function PokemonCard({
+const PokemonCard: React.FC<cardProps> = ({
   id,
   name,
   imgURL,
-
-  ability,
   baseColor,
   type,
-}: cardProps) {
-  // @ts-ignore
-  let types: string = type[0].type.name;
-  const navigate = useNavigate();
-
-  const handleCardClick = (id: number) => {
-    console.log("Navigating to /pokemon/" + id);
-
-    navigate(`/pokemon/${id}`);
-  };
+  onClick,
+}) => {
+  const types = type?.[0]?.type?.name ?? "";
 
   return (
     <div
-      onClick={() => handleCardClick(id)}
-      className="w-1/6  bg-slate-200 m-6 p-4 rounded-sm flex flex-col items-center h-1/3 justify-evenly "
+      onClick={() => onClick(id)}
```

Comment:
Since the pokemon object is also available on the map function on the search component, it would be better to handle onclick there itself.

```suggestion
      onClick={cardOnClick}
```